normandy/recipes/tests/test_filters.py

```diff
@@ -149,6 +155,39 @@ def test_generates_jexl_error_on_bad_comparison(self):
             filter.to_jexl()
 
 
+class TestWindowsVersionFilter(FilterTestsBase):
+    def create_basic_filter(self, value="6.1", comparison="equal"):
+        WindowsVersionFactory(slug="6.1")
+
+        return WindowsVersionFilter.create(value=value, comparison=comparison)
+
+    @pytest.mark.parametrize(
+        "comparison,symbol",
+        [
+            ("equal", "=="),
+            ("greater_than", ">"),
+            ("greater_than_equal", ">="),
+            ("less_than", "<"),
+            ("less_than_equal", "<="),
+        ],
+    )
+    def test_generates_jexl_number_ops(self, comparison, symbol):
+        filter = self.create_basic_filter(comparison=comparison)
+        assert (
+            filter.to_jexl()
+            == f"(normandy.os.isWindows && normandy.os.windowsVersion {symbol} 6.1)"
+        )
+
+    def test_generates_jexl_error_on_bad_comparison(self):
+        filter = self.create_basic_filter(comparison="typo")
+        with pytest.raises(serializers.ValidationError):
+            filter.to_jexl()
+
+    def test_generates_jexl_error_on_bad_version(self):
+        with pytest.raises(AssertionError):
```

Comment:
I haven't been able to find it. What is raising this AssertionError? I can only find `ValidationError`s that should be raised from here.

Reply:
It's coming from this place in the BaseFilter: https://github.com/mozilla/normandy/blob/master/normandy/recipes/filters.py#L35

the `validate_value` method is causing `ob.is_valid( )` to be false there.

Reply:
Ah, ok. Thanks for showing me that. I also verified that this comes through in HTTP requests nicely:

```shell
$ $ edi recipes create
    -A 'Insecure mcooper@mozilla.com'
    --server local
    --cert ~/src/normandy/etc/ssl/normandy_dev.crt 
	'{"action_id":4,"arguments":"{\"message\":\"test\"}","name":"test","
filter_object":[{"type":"windows_version","value":"asdf","comparison":"less"}]}'

Error: HTTP Error Code 400: {"filter_object":{"0":{"value":["Unrecognized windows version slug 'asdf'"]}}}
```